java-checks/src/main/java/org/sonar/java/checks/ThreadRunCheck.java

```diff
@@ -50,7 +54,19 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
     if (parent != null && THREAD_RUN_METHOD_MATCHER.matches((MethodTree) parent)) {
       return;
     }
-    reportIssue(ExpressionUtils.methodName(mit), "Call the method Thread.start() to execute the content of the run() method in a dedicated thread.");
+
+    IdentifierTree methodName = ExpressionUtils.methodName(mit);
+    QuickFixHelper.newIssue(context)
+      .forRule(this)
+      .onTree(methodName)
+      .withMessage("Call the method Thread.start() to execute the content of the run() method in a dedicated thread.")
+      .withQuickFix(() -> computeQuickFix(methodName))
+      .report();
   }
 
+  private static JavaQuickFix computeQuickFix(IdentifierTree methodName) {
+    return JavaQuickFix.newQuickFix("Call method Thread.start()")
```

Comment:
While it's true that the raised issue message is `"Call the method Thread.start() to execute..."` , I feel like the quick fix message could describe a little bit better what it will actually do on the code when applied, maybe something along the lines of:
```suggestion
    return JavaQuickFix.newQuickFix("Replace the .run() call with .start() call")
```
Or something like that, maybe you can come up with something else that sounds better!

Reply:
You are right.
What about 'Replace run() with start()'?

Reply:
Yeah sounds good to me, after this I feel we're good to go! Nice job!